apis/v1alpha2/validation/httproute.go

```diff
@@ -166,3 +166,31 @@ func validateHTTPPathMatch(path *gatewayv1a2.HTTPPathMatch, fldPath *field.Path)
 	}
 	return allErrs
 }
+
+// validateHTTPRouteFilterTypeMatchesValues validates that the filter type matches the filter value.
+func validateHTTPRouteFilterTypeMatchesValues(filter *gatewayv1a2.HTTPRouteFilter, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+
+	switch filter.Type {
+	case gatewayv1a2.HTTPRouteFilterExtensionRef:
+		if filter.ExtensionRef == nil {
+			return append(errs, field.Invalid(path.Child("filters"), path, "type of filter must match type of value"))
```

Comment:
```suggestion
			errs = append(errs, field.Invalid(path.Child("filters"), path, "type of filter must match type of value"))
```

Follow this pattern in the function, so that returning errs makes more sense.

Reply:
Oh yes, good catch. Thanks.